x-pack/plugins/reporting/server/routes/jobs.ts

```diff
@@ -50,7 +56,7 @@ export function registerJobInfoRoutes(reporting: ReportingCore) {
         page: queryPage = '0',
         size: querySize = '10',
         ids: queryIds = null,
-      } = req.query as ListQuery;
+      } = req.query as ListQuery; // NOTE: type inference is not working here. userHandler breaks it?
```

Comment:
I would like to understand why the type of `req.query` can not be inferred from the validation. I didn't take much time to look into it.

Reply:
I'd like to know as well. I spent a good deal of time trying to get types to pass through with the higher order handler (https://github.com/elastic/kibana/blob/master/x-pack/plugins/reporting/server/routes/lib/authorized_user_pre_routing.ts#L15), but there's some interesting effects it might have trying to get type information to pass through properly. This was a pretty big change in our codebase to get everything working properly, as a lot of our utilities relied on augmented Hapi request objects since we used the `pre` handlers fairly extensively. I can take a look at this since I wrote the wrapper handler. Unfortunately nothing is jumping out at me since I pretty much copy/pasted the migration docs for this "higher-order" handler.